tpu-client/src/tpu_client.rs

```diff
@@ -77,6 +77,10 @@ impl TpuClient {
         self.invoke(self.tpu_client.try_send_transaction(transaction))
     }
 
+    pub fn try_send_transaction_batch(&self, transactions: &[Transaction]) -> TransportResult<()> {
```

Comment:
There should be a function TpuConnection::par_serialize_and_send_transaction_batch that should do what you need (serialize the transactions and batch send them). IMHO it should be possible to use already existing functions in TpuConnection, and the newly added functions should be unnecessary (I hope).

Reply:
Whoops nevermind, misread where the changes were being made. The way things are currently is fine. The one thing I kind of don't really like that the serialization takes place on a single thread on a runtime... perferably the serialization could take place outside an async function and with parallelism as in TpuConnection::par_serialize_and_send_transaction_batch if possible

Reply:
Thanks @ryleung-solana for the review. I have updated the PR to use par_iter to serialize the transactions.